### PR TITLE
docs/python/mobile: Updates and cleanups

### PR DESCRIPTION
**Pull request by phlax**
- Use bazel for mobile docs build
- Remove (most) vendored sphinx packages
- Remove remaining host python utils

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@abeyad im reasonably confident that this doesnt break mobile docs publishing, but will monitor when it lands to make sure

File: .github/workflows/mobile-docs.yml
Comment (abeyad) [inline .github/workflows/mobile-docs.yml:37]: i haven't run mobile docs in a very long time, so more just curious if you tried this command locally and it worked?

**Review by abeyad — COMMENTED** [review]
so after this PR, what's the correct way to generate mobile docs?  i.e. what's the exact command we would run if we wanted to see what doc changes look like locally?

File: .github/workflows/mobile-docs.yml
Comment (phlax) [reply .github/workflows/mobile-docs.yml:37]: yeah - ive been running quite a bit for this PR - but previously also to test sphinx updates - one of the  motivations for this PR is to make that easier to maintain

**Comment by phlax** [comment]
> so after this PR, what's the correct way to generate mobile docs? i.e. what's the exact command we would run if we wanted to see what doc changes look like locally?

directly on host (assuming bazel):

```console
$ cd mobile
$ docs/build.sh
```

using the envoy build container:

```console
$ ./ci/run_envoy_docker.sh 'cd mobile && docs/build.sh'
```

i would like to integrate the ci here with envoys docs - it would make things easier and we would get mobile docs publishing in ci

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

I'll let @abeyad have a final approval on these changes.

**Comment by phlax** [comment]
> what's the exact command we would run if we wanted to see what doc changes look like locally?

heres a suggestion - assuming above has been run

```console
$ docker run --rm -it -p 8080:80 -v "${PWD}/generated/docs:/usr/share/nginx/html" nginx  
```

File: .github/workflows/mobile-docs.yml
Comment (phlax) [reply .github/workflows/mobile-docs.yml:37]: good shout here

i just reran it and realized i introduced an error at the end - checking with a local file server highlighed this - fixing now ...

**Comment by abeyad** [comment]
> > so after this PR, what's the correct way to generate mobile docs? i.e. what's the exact command we would run if we wanted to see what doc changes look like locally?
> 
> directly on host (assuming bazel):
> 
> ```
> $ cd mobile
> $ docs/build.sh
> ```
> 
> using the envoy build container:
> 
> ```
> $ ./ci/run_envoy_docker.sh 'cd mobile && docs/build.sh'
> ```
> 
> i would like to integrate the ci here with envoys docs - it would make things easier and we would get mobile docs publishing in ci

Thanks for the explanation!  Is there somewhere in our docs (I know, it's very meta :) ) we can put this so Envoy engineers can know?

**Review by abeyad — APPROVED** [review]
Thanks for doing this, looks great!

**Comment by phlax** [comment]
> Thanks for the explanation! Is there somewhere in our docs (I know, it's very meta :) ) we can put this so Envoy engineers can know?

probs this would be useful for envoy also - but can we address in a follow up ?

**Comment by abeyad** [comment]
> > Thanks for the explanation! Is there somewhere in our docs (I know, it's very meta :) ) we can put this so Envoy engineers can know?
> 
> probs this would be useful for envoy also - but can we address in a follow up ?

SGTM!

**Comment by abeyad** [comment]
> > Thanks for the explanation! Is there somewhere in our docs (I know, it's very meta :) ) we can put this so Envoy engineers can know?
> 
> ~probs this would be useful for envoy also - but can we address in a follow up ?~
> 
> reread q properly - its there in the mobile/docs/README.md i think

we probably need to update https://github.com/envoyproxy/envoy/blob/2e9bf6bf4396e45de899a5c962fcb1e2579697cb/mobile/docs/README.md?plain=1#L12 to the command in https://github.com/envoyproxy/envoy/pull/29606/files#diff-50435bfc45f7a52382eb14cf51fd3e526911f5139a2d151da0c07372f457ec65R37, no?

**Comment by phlax** [comment]
> we probably need to update 

we can - i guess the difference now is that it assumes bazel

ill add something ...

**Comment by phlax** [comment]
> ill add something ...

i dont want to restart ci - ill update the README with both above suggestions in follow up

**Comment by abeyad** [comment]
> > ill add something ...
> 
> i dont want to restart ci - ill update the README with both above suggestions in follow up

SGTM

**Pull request merged**
